File: restaurant/controller/table_controller.py
from service.table_service import TableService

class TableController:
    """Controller xử lý các yêu cầu liên quan đến bàn"""
    
    def __init__(self, table_service: TableService):
        self.table_service = table_service
    
    def list_all_tables(self):
        """Lấy danh sách tất cả bàn"""
        tables = self.table_service.get_all_tables()
        return tables
    
    def list_available_tables(self):
        """Lấy danh sách bàn trống"""
        tables = self.table_service.get_available_tables()
        return tables
    
    def get_table_info(self, table_id: int):
        """Lấy thông tin chi tiết của một bàn"""
        table = self.table_service.get_table(table_id)
        return table
    
    def book_table(self, table_id: int) -> bool:
        """Đặt bàn (đánh dấu bàn đã có người)"""
        return self.table_service.occupy_table(table_id)
    
    def free_table(self, table_id: int) -> bool:
        """Giải phóng bàn"""
        return self.table_service.release_table(table_id)

File: backend/src/main/java/com/restaurant/controller/TableController.java
package com.restaurant.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping; // [QUAN TRỌNG] Import Enum
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.restaurant.model.entity.Table;
import com.restaurant.model.enums.TableStatus;
import com.restaurant.service.TableService;

@RestController
@RequestMapping("/api/tables")
@CrossOrigin(origins = "*")
public class TableController {

    private final TableService tableService;

    @Autowired
    public TableController(TableService tableService) {
        this.tableService = tableService;
    }

    // 1. Lấy danh sách bàn
    @GetMapping
    public List<Table> getAllTables() {
        return tableService.getAllTables();
    }

    // 2. Thêm bàn mới
    // [SỬA LỖI] Bỏ "/add" để khớp với Frontend (POST /api/tables)
    @PostMapping 
    public ResponseEntity<Table> addTable(@RequestBody Table table) {
        return ResponseEntity.ok(tableService.saveTable(table));
    }

    // 3. Cập nhật trạng thái
    // [SỬA LỖI] Chuyển đổi String "occupied" -> Enum TableStatus.OCCUPIED
    @PutMapping("/{id}/status")
    public ResponseEntity<Table> updateTableStatus(@PathVariable int id, @RequestParam String status) {
        try {
            // Chuyển chuỗi sang Enum (không phân biệt hoa thường)
            TableStatus tableStatus = TableStatus.valueOf(status.toUpperCase());
            return ResponseEntity.ok(tableService.updateStatus(id, tableStatus));
        } catch (IllegalArgumentException e) {
            // Nếu gửi sai status (vd: "abc") -> Báo lỗi Bad Request
            return ResponseEntity.badRequest().build();
        }
    }

    // 4. Xóa bàn
    @DeleteMapping("/{id}")
    public ResponseEntity<String> deleteTable(@PathVariable int id) {
        tableService.deleteTable(id);
        return ResponseEntity.ok("Đã xóa bàn ID: " + id);
    }
}